duckduckgo/Android
# Feature/ui performance

Task/Issue URL: https://app.asana.com/0/414730916066338/705937877094329
Tech Design URL: 
CC: 

**Description**:
Breaks apart the `BrowserTabViewModel`'s view state into smaller view states so that upon each change to an item in the view state, only related UI elements are updated. 

**Steps to test this PR**:
1. _Vague instructions:_ Generally use the browser - kick the tyres and make sure no regressions were introduced in the process
1. _[optionally]_ Verify the Play Store pre-release performance results are positive


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
